contracts/base/FallbackManager.sol

```diff
@@ -18,6 +18,8 @@ abstract contract FallbackManager is SelfAuthorized {
      *  @param handler contract to handle fallback calls.
      */
     function internalSetFallbackHandler(address handler) internal {
+        require(handler != address(this), "GS400");
```

Comment:
I implemented the check here because we call the internal function directly in the setup method. I assume it was done for gas savings of a `ChangeFallbackHandler` event emission